encoding/cbf/cbf_codec/encode.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cbf_codec
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/cbf/common_codec"
+)
+
+const VERSION = byte(0x01)
+
+// An Encoder converts Cadence values into custom-encoded bytes.
+type Encoder struct {
```

Comment:
maybe more descriptive field names than w for writer and r for reader? they aren't too obvious in usage contexts